contracts/src/SP1Helios.sol

```diff
@@ -122,6 +131,11 @@ contract SP1Helios {
             revert PreviousHeadNotSet(head);
         }
 
+        // Check if the head being proved against is more than 2 weeks old
+        if (latestHead - head > MAX_ALLOWED_SLOTS_BEHIND) {
```

Comment:
Not sure if this is exactly the check we need. I think we need to compare head against the _current time_, right? Because latestHead could also be old, no?

Reply:
I am not sure we have knowledge of what the timestamp of the head is though right?